cmake/nokia_ngage.cmake

```diff
@@ -0,0 +1,133 @@
+cmake_minimum_required(VERSION 3.16)
+
+enable_language(CXX)
+
+set(GCC_MODULE_COMN_DEFS -D__SYMBIAN32__ -D__GCC32__ -D__EPOC32__ -D__MARM__ -D__MARM_ARMI__)
+set(GCC_MODULE_MODE_DEFS -DNDEBUG -D_UNICODE)
+set(GCC_MODULE_DEFS ${GCC_MODULE_COMN_DEFS} ${GCC_MODULE_MODE_DEFS})
+
+file(GLOB SDL3_sources
+  "${SDL3_SOURCE_DIR}/src/*.c"
+  "${SDL3_SOURCE_DIR}/src/atomic/*.c"
+  "${SDL3_SOURCE_DIR}/src/audio/*.c"
+  "${SDL3_SOURCE_DIR}/src/audio/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/audio/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/core/*.c"
+  "${SDL3_SOURCE_DIR}/src/core/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/core/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/cpuinfo/*.c"
+  "${SDL3_SOURCE_DIR}/src/dynapi/*.c"
+  "${SDL3_SOURCE_DIR}/src/events/*.c"
+  "${SDL3_SOURCE_DIR}/src/file/*.c"
+  "${SDL3_SOURCE_DIR}/src/file/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/filesystem/posix/*.c"
+  "${SDL3_SOURCE_DIR}/src/hidapi/*.c"
+  "${SDL3_SOURCE_DIR}/src/io/*.c"
+  "${SDL3_SOURCE_DIR}/src/io/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/libm/s_isinff.c"
+  "${SDL3_SOURCE_DIR}/src/libm/s_isnanf.c"
+  "${SDL3_SOURCE_DIR}/src/locale/*.c"
+  "${SDL3_SOURCE_DIR}/src/locale/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/main/*.c"
+  "${SDL3_SOURCE_DIR}/src/main/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/main/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/misc/*.c"
+  "${SDL3_SOURCE_DIR}/src/misc/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/power/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/stdlib/*.c"
+  "${SDL3_SOURCE_DIR}/src/storage/*.c"
+  "${SDL3_SOURCE_DIR}/src/storage/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/thread/*.c"
+  "${SDL3_SOURCE_DIR}/src/thread/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/time/*.c"
+  "${SDL3_SOURCE_DIR}/src/time/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/time/unix/*.c"
+  "${SDL3_SOURCE_DIR}/src/timer/*.c"
+  "${SDL3_SOURCE_DIR}/src/timer/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/tray/*.c"
+  "${SDL3_SOURCE_DIR}/src/tray/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/yuv2rgb/*.c")
+
+add_library(${PROJECT_NAME} STATIC ${SDL3_sources})
+
+target_include_directories(
+  ${PROJECT_NAME}
+  PUBLIC
+  "${SDL3_SOURCE_DIR}/include"
+  PRIVATE
+  "${SDL3_SOURCE_DIR}/src"
+  "${SDL3_SOURCE_DIR}/include/build_config")
+
+target_compile_definitions(
+  ${PROJECT_NAME}
+  PUBLIC
+  SDL_STATIC_LIB
+  ${GCC_MODULE_DEFS})
+
+set(test_static_libs
+  ${CMAKE_CURRENT_BINARY_DIR}/libSDL3.a)
+
+set(test_libs
+  ${CMAKE_CURRENT_BINARY_DIR}/libSDL3.a
+  ${EPOC_LIB}/NRenderer.lib
+  ${EPOC_LIB}/3dtypes.a
+  ${EPOC_LIB}/cone.lib
+  ${EPOC_PLATFORM}/gcc/lib/gcc-lib/arm-epoc-pe/2.9-psion-98r2/libgcc.a
+  ${EPOC_PLATFORM}/ngagesdk/lib/gcc/arm-epoc-pe/4.6.4/libgcc_ngage.a
```

Comment:
Perhaps add these to  [`CMAKE_C_STANDARD_LIBRARIES`]https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_STANDARD_LIBRARIES.html] in your CMake toolchain file?

Reply:
Sounds good! I hope it does not affect the linking order.

Reply:
Does not work since these libraries are also required in a custom-defined linking process.